src/Confluent.SchemaRegistry.Serdes.Avro/SpecificSerializerImpl.cs

```diff
@@ -33,23 +34,61 @@ namespace Confluent.SchemaRegistry.Serdes
 {
     internal class SpecificSerializerImpl<T> : IAvroSerializerImpl<T>
     {
+        internal class SpecificSerializerSchemaData
+        {
+            private string writerSchemaString;
+            private global::Avro.Schema writerSchema;
+
+            /// <remarks>
+            ///     A given schema is uniquely identified by a schema id, even when
+            ///     registered against multiple subjects.
+            /// </remarks>
+            private int? writerSchemaId;
+
+            private SpecificWriter<T> avroWriter;
+            
+            private HashSet<string> subjectsRegistered = new HashSet<string>();
+
+            public HashSet<string> SubjectsRegistered
+            {
+                get => subjectsRegistered;
+                set => subjectsRegistered = value;
+            }
+
+            public string WriterSchemaString
+            {
+                get => writerSchemaString;
+                set => writerSchemaString = value;
+            }
+
+            public Avro.Schema WriterSchema
+            {
+                get => writerSchema;
+                set => writerSchema = value;
+            }
+
+            public int? WriterSchemaId
+            {
+                get => writerSchemaId;
+                set => writerSchemaId = value;
+            }
+
+            public SpecificWriter<T> AvroWriter
+            {
+                get => avroWriter;
+                set => avroWriter = value;
+            }
+        }
+
         private ISchemaRegistryClient schemaRegistryClient;
         private bool autoRegisterSchema;
+        private bool _isMultipleSchemas = false;
         private int initialBufferSize;
         private SubjectNameStrategyDelegate subjectNameStrategy;
 
-        private string writerSchemaString;
-        private global::Avro.Schema writerSchema;
-
-        /// <remarks>
-        ///     A given schema is uniquely identified by a schema id, even when
-        ///     registered against multiple subjects.
-        /// </remarks>
-        private int? writerSchemaId;
+        private Dictionary<Type, SpecificSerializerSchemaData> schemaData =
```

Comment:
I think this should be called `multiSchemaData`, and single schema data should be held in another field `schemaData`. Then you also don't need `_isMultipleSchemas`